autogpts/autogpt/autogpt/agents/components.py

```diff
@@ -0,0 +1,40 @@
+from typing import Callable, Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class Single(Generic[T]):
+    """A wrapper for a single result (non-pipeline) of a component function."""
+
```

Comment:
Needed (hopefully temporarily) to handle copying of component/pipeline result data effectively. `Single` values are effectively for non-pipeline components.